### PR TITLE
feat(url): Add `quirks::internal_components`

### PR DESCRIPTION
**Pull request by littledivy**
Supersedes #762 with added doc comments

### CODE REVIEW

**Review by lucacasonato — COMMENTED** [review]
This looks good to me functionality wise. Please add a couple of tests.

@valenting do you have concerns exposing this internal structure in a semi-public API? Exposing this info would allow us to significantly speed up our `URL` JS global in Deno.

**Comment by valenting** [comment]
I would prefer feature-gate this API instead. If that's not possible I can live with it. 🤷‍♂️

File: url/Cargo.toml [hunk 1]
```diff
@@ -38,6 +38,7 @@ default = ["idna"]
 # UNSTABLE FEATURES (requires Rust nightly)
 # Enable to use the #[debugger_visualizer] attribute.
 debugger_visualizer = []
+expose_internals = []
```
[the remark this answers is not in the document]
Comment (littledivy) [reply url/Cargo.toml:42]: Done

**Comment by valenting** [comment]
LGTM. @lucacasonato approve and merge at your convenience.

**Review by lucacasonato — APPROVED** [review]
LGTM, thanks.

**Pull request merged**